EventListener/ControllerArgumentOptionsListener.php

```diff
@@ -0,0 +1,138 @@
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Parses annotation blocks located in controller classes to extract @Arg ones.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class ControllerArgumentOptionsListener implements EventSubscriberInterface
+{
+    /**
+     * @var Reader
+     */
+    private $reader;
+    private $metadataFactory;
+    private $argumentOptions;
+    private $docBlockFactory;
+
+    public function __construct(Reader $reader, ArgumentMetadataFactory $metadataFactory = null, array $argumentOptions)
+    {
+        $this->reader = $reader;
+        $this->metadataFactory = $metadataFactory;
+        $this->argumentOptions = $argumentOptions;
+    }
+
+    public function onKernelController(FilterControllerEvent $event)
+    {
+        $controller = $event->getController();
+
+        if (!is_array($controller) && method_exists($controller, '__invoke')) {
+            $controller = array($controller, '__invoke');
+        }
+
+        if (!is_array($controller)) {
+            return;
+        }
+
+        $className = class_exists('Doctrine\Common\Util\ClassUtils') ? ClassUtils::getClass($controller[0]) : get_class($controller[0]);
+        $method = new \ReflectionMethod($className, $controller[1]);
+        $annotations = $this->reader->getMethodAnnotations($method);
+        $context = $className.'::'.$controller[1].'()';
+        $argumentMetadata = $this->metadataFactory->createArgumentMetadata($controller);
+
+        $parameters = array();
+        foreach ($method->getParameters() as $parameter) {
+            $parameters[$parameter->getName()] = true;
+
+            $options = array();
+            foreach ($annotations as $annotation) {
+                if (!$annotation instanceof Arg) {
+                    continue;
+                }
+
+                if ($annotation->getArg() !== $parameter->getName()) {
+                    continue;
+                }
+
+                if (null === $argumentMetadata[$parameter->getName()]->getType()) {
+                    throw new LogicException(sprintf('Class type hint on argument "$%s" is required when defining an @Arg annotation.', $parameter->getName()));
+                }
+
+                $options = $annotation->getOptions();
+            }
+
+            if (null === $argumentMetadata[$parameter->getName()]->getType()) {
+                continue;
+            }
+
+            try {
+                $options = $this->resolveOptions($event->getRequest(), $parameter->getName(), $argumentMetadata[$parameter->getName()], $options);
+            } catch (OptionsResolverExceptionInterface $e) {
+                throw new LogicException(sprintf('Wrong @Arg configuration for argument "%s" on "%s".', $name, $context), 0, $e);
+            }
+
+            $event->getRequest()->attributes->set($parameter->getName().'_options', $options);
```

Comment:
Ah nice leveraging the use of the old setup here, that would prevent them from being executed unless they set a higher priority on their own converter. This should probably be documented as pitfall if they decide to use a higher priority and still try to use `@Arg`.